enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,437 @@
+As a common baseline, a new top level section will be added to the
+configuration:
+```yaml
+apiServer:
+   ...
+# new from here.
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+  expose:
+    - <IP address/NIC> # Defaults to the NICs of the node
+```
+
+### Enable/disable the router
+The following configuration is proposed:
+```yaml
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+```
+
+With this option MicroShift will decide whether it should create the router
+upon starting. This includes not just the pod, but all the associated resources
+that come along with it: namespace, services, configmaps, etc.
+
+When setting the option to `Disabled`, the next MicroShift restart should delete
+all the router related resources, including APIs and controllers.
+
+Setting the option to `Enabled` will deploy the router, while `Disabled` will disable
+it. Default will be `Enabled`.
+
+### Listening ports
+The following configuration is proposed:
+```yaml
+ingress:
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+```
+
+In order to allow this configuration and many other advantages, the
+router service shall be exposed using `LoadBalancer` type. See the
+Design Details section for more information.
+
+#### Firewalling ports
+Using `LoadBalancer` service type prevents the usage of firewalld to block
+access to ports. Since this is a feature that users may require for auditing,
+an alternative is provided.
+
+The `NetworkPolicy` resource fits the needs for blocking traffic to services
+and pods. Policies can also be configured to allow specific IP addresses and
+provide a high degree of customization for network traffic. They use the CNI
+instead of host level settings, providing greater flexibility and being self
+contained within MicroShift.
+
+Users in need of blocking traffic shall use the `NetworkPolicy` resource. A
+section in the documentation will be added to describe how to do this.
+
+### Configure IPs where the router listens
+The following configuration is proposed:
+```yaml
+ingress:
+  expose:
+    - <IP address/NIC> # Defaults to the NICs of the node
+```
+
+As we have seen in previous sections, the use of `LoadBalancer` makes ovnk
+configure iptables rules to expose the service outside of the cluster.
+Building on [LoadBalancer controller enhancement](loadbalancer-service-support.md),
+the service is exposed using the node IP and configured ports.
+
+All the IPs where the router is exposed need to be part of the `LoadBalancer`
+service status definition. In order to provide more flexibility, the
+configuration options are able to take IP addresses and/or NIC names in the
+system. Using NIC names abstracts away IP changes, in cases where this
+happens.
+
+As a default value, MicroShift will take the NIC names for each of the default
+routes found in the system.
+
+### Allow specific IP addresses
```

Comment:
We talked about removing the discussions of using NetworkPolicy, since that is an existing API and doesn't represent a design change in the system. We will document how to use them elsewhere. You could _mention_ that they will be used to deal with the firewall-like behaviors, but we don't need a lot of detail about it.

Reply:
So the example can also be removed? I tried to capture that in the first 2 paragraphs and then provide an example for guidance/clarity.

Reply:
Done